### PR TITLE
All Tickets Page > Provider Hooks to Parent Class and Add Column Filters

### PR DESCRIPTION
**Pull request by codingmusician**
### 🎫 Ticket
[ETP-938]

### 🗒️ Description
We are moving all the provider hooks for the All Tickets page to the parent `Tribe__Tickets__Tickets` class. We also added filters to each of the All Tickets table columns.

We're also changing any instance of "All Tickets" to "Admin Tickets", with exception of the menu item name and page title. So any classes, functions or methods using `all-tickets` or `all_tickets` should be renamed to `admin-tickets` or `admin_tickets`, respectively.

### 🎥 Artifacts 
Shows how WooCommerce can filter the Sold and Sales columns, eventhough the orders are "Pending":
image

### ✔️ Checklist
- [ ] Ran `npm run changelog` to add changelog file(s). More info [here](https://docs.theeventscalendar.com/developer/git/changelogs/#process)
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [x] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [x] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [x] Add your PR to the project board for the release.

[ETP-938]: https://stellarwp.atlassian.net/browse/ETP-938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Admin/Tickets/List_Table.php [hunk 1]
```diff
@@ -377,7 +390,18 @@ public function column_name( $item ): string {
...
 
-		return $template->template( 'tickets/column/name', $context, false );
+		$name = $template->template( 'tickets/column/name', $context, false );
+
+		/**
+		 * Filters the name for the All Tickets Table.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $name The name for the All Tickets Table.
```
Comment (redscar) [inline src/Tickets/Admin/Tickets/List_Table.php:400]: Missing the `$item` as a param. I noticed in a few other filters below you are also missing the `$item` param.

File: src/Tickets/Admin/Tickets/List_Table.php [hunk 2]
```diff
@@ -377,7 +390,18 @@ public function column_name( $item ): string {
...
+		 *
+		 * @return string
+		 */
+		return apply_filters( 'tec_tickets_all_tickets_table_column_name', $name, $item );
```
Comment (Camwyn) [inline src/Tickets/Admin/Tickets/List_Table.php:404]: We'd typically do two of these, the second one would have the name in the filter, i.e.
`tec_tickets_all_tickets_table_column_name_{$name}`

**Review by bordoni — CHANGES_REQUESTED** [review]
Avoid all references to `All Tickets` even on filters or function names, lets be consistent with our class names.

File: src/Tickets/Admin/Tickets/List_Table.php [hunk 3]
```diff
@@ -497,12 +543,23 @@ public function column_start( $item ): string {
...
+		/**
+		 * Filters the start date for the All Tickets Table.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $start The start date for the All Tickets Table.
+		 *
+		 * @return string
+		 */
+		return apply_filters( 'tec_tickets_all_tickets_table_column_start', $start, $item );
```
Comment (Camwyn) [inline src/Tickets/Admin/Tickets/List_Table.php:562]:
```suggestion
		return apply_filters( 'tec_tickets_all_tickets_table_column_start_date', $start, $item );
```
Let's be clear in the filter that it's the start _date_

File: src/Tickets/Admin/Tickets/List_Table.php [hunk 4]
```diff
@@ -522,12 +579,23 @@ public function column_end( $item ): string {
...
+		/**
+		 * Filters the end date for the All Tickets Table.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $end The end date for the All Tickets Table.
+		 *
+		 * @return string
+		 */
+		return apply_filters( 'tec_tickets_all_tickets_table_column_end', $end, $item );
```
Comment (Camwyn) [inline src/Tickets/Admin/Tickets/List_Table.php:598]:
```suggestion
		return apply_filters( 'tec_tickets_all_tickets_table_column_end_date', $end, $item );
```
Let's be clear in the filter that it's the end _date_

File: src/Tribe/Tickets.php [hunk 5]
```diff
@@ -4720,6 +4730,36 @@ public function deactivate(): void {
...
+		 *
+		 * @return string[] The filtered list of event meta keys.
+		 */
+		public function filter_all_tickets_table_event_meta_keys( $event_meta_keys ) {
+			$event_meta_keys[ $this->ticket_object ] = $this->get_event_key();
+
+			return $event_meta_keys;
+		}
+
 		// @codingStandardsIgnoreEnd
```
Comment (Camwyn) [inline src/Tribe/Tickets.php:4763]: These functions are inside the `// @codingStandardsIgnore` for the deprecated methods - move them up and out of that please. (and any others that were put there inadvertently)

**Review by Camwyn — CHANGES_REQUESTED** [review]

**Review by Camwyn — APPROVED** [review]

**Review by bordoni — APPROVED** [review]

**Pull request merged**